features/storage/TESTS/kvstore/general_tests/main.cpp

```diff
@@ -0,0 +1,961 @@
+    int res = kvstore->set_start(&handle, key, new_data_size, 0);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->set_add_data(handle, data, new_data_size - 1);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->set_add_data(handle, data, data_size);
+    TEST_ASSERT_EQUAL(MBED_ERROR_INVALID_SIZE, res);
+
+    kvstore->reset();
+}
+
+//set different data_sizes chunks of data in the same transaction
+static void set_add_data_set_different_data_size_in_same_transaction()
+{
+    KVStore::set_handle_t handle;
+
+    char new_data[] = "new_data_tests";
+    size_t new_data_size = 15;
+    
+    int res = kvstore->set_start(&handle, key, new_data_size, 0);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->set_add_data(handle, new_data, new_data_size - 5);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->set_add_data(handle, new_data, 0);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->set_add_data(handle, new_data + (new_data_size - 5), 5);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->set_finalize(handle);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->get(key, buffer, buffer_size, &actual_size, 0);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    TEST_ASSERT_EQUAL_STRING(new_data, buffer);
+
+    kvstore->reset();
+}
+
+//set key value five Kbyte size
+static void set_add_data_set_key_value_five_Kbytes()
+{
+    KVStore::set_handle_t handle;
+
+    size_t new_data_size = 5000;
+    size_t temp_buf[50] = {};
+    unsigned int i = 0;
+
+    int res = kvstore->set_start(&handle, key, new_data_size, 0);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    for (i = 0; i < (new_data_size / sizeof(temp_buf)); i++) {
+        memset(temp_buf, '*', sizeof(temp_buf));
+        res = kvstore->set_add_data(handle, temp_buf, sizeof(temp_buf));
+    }
+
+    res = kvstore->set_finalize(handle);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->get(key, temp_buf, sizeof(temp_buf), &actual_size, 0);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    kvstore->reset();
+}
+
+//set_add_data without set_start
+static void set_add_data_without_set_start()
+{
+    KVStore::set_handle_t handle;
+
+    int res = kvstore->set_add_data(handle, data, data_size);
+    TEST_ASSERT_EQUAL(MBED_ERROR_INVALID_ARGUMENT, res);
+}
+
+/*----------------set_finalize()------------------*/
+
+//set_finalize without set_start
+static void set_finalize_without_set_start()
+{
+    KVStore::set_handle_t handle;
+
+    int res = kvstore->set_finalize(handle);
+    TEST_ASSERT_EQUAL(MBED_ERROR_INVALID_ARGUMENT, res);
+}
+
+static void set_finalize_right_after_set_start()
+{
+    KVStore::set_handle_t handle;
+
+    int res = kvstore->set_start(&handle, key, 0, 0);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+
+    res = kvstore->set_finalize(handle);
+    TEST_ASSERT_EQUAL(MBED_SUCCESS, res);
+}
+
+
+/*----------------setup------------------*/
+
+Case cases[] = {
```

Comment:
It is better to define a failure handler that will make the tests continue. You have a lot of tests here, and it's better to have all of them run, even if one ore more fail.